pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -103,54 +210,350 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
-			Name:      cr.Name + "-pod",
-			Namespace: cr.Namespace,
-			Labels:    labels,
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Name:    "busybox",
-					Image:   "busybox",
-					Command: []string{"sleep", "3600"},
-				},
-			},
-		},
+	mcpMutated, created, err := r.getMutatedMachineConfigPool(mcp)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated machine config
+	mc, err := machineconfig.New(r.assetsDir, profile)
+	if err != nil {
+		return nil, err
+	}
+	if err := controllerutil.SetControllerReference(profile, mc, r.scheme); err != nil {
+		return nil, err
+	}
+	mcMutated, err := r.getMutatedMachineConfig(mc)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated feature gate
+	fg := featuregate.NewLatencySensitive()
+	// TOOD: uncomment once https://bugzilla.redhat.com/show_bug.cgi?id=1788061 fixed
+	// if err := controllerutil.SetControllerReference(profile, fg, r.scheme); err != nil {
+	// 	return err
+	// }
+	fgMutated, err := r.getMutatedFeatureGate(fg)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated kubelet config
+	kc, err := kubeletconfig.New(profile)
+	if err != nil {
+		return nil, err
+	}
+	if err := controllerutil.SetControllerReference(profile, kc, r.scheme); err != nil {
+		return nil, err
+	}
+	kcMutated, err := r.getMutatedKubeletConfig(kc)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated network latency tuned
+	networkLatencyTuned, err := tuned.NewNetworkLatency(r.assetsDir)
+	if err != nil {
+		return nil, err
+	}
+	if err := controllerutil.SetControllerReference(profile, networkLatencyTuned, r.scheme); err != nil {
+		return nil, err
+	}
+	networkLatencyTunedMutated, err := r.getMutatedTuned(networkLatencyTuned)
+	if err != nil {
+		return nil, err
+	}
+
+	// get mutated real time kernel tuned
+	realTimeKernelTuned, err := tuned.NewWorkerRealTimeKernel(r.assetsDir, profile)
+	if err != nil {
+		return nil, err
+	}
+	if err := controllerutil.SetControllerReference(profile, realTimeKernelTuned, r.scheme); err != nil {
+		return nil, err
+	}
+	realTimeKernelTunedMutated, err := r.getMutatedTuned(realTimeKernelTuned)
+	if err != nil {
+		return nil, err
+	}
+
+	// does not update any resources, if it no changes to relevant objects and just continue to the status update
+	if mcpMutated == nil &&
+		mcMutated == nil &&
+		kcMutated == nil &&
+		fgMutated == nil &&
+		networkLatencyTunedMutated == nil &&
+		realTimeKernelTunedMutated == nil {
+		// if nothing needed to be updated and machine config pool paused, unpause it
+		mcpUpdate, err := r.getMachineConfigPool(mcp.Name)
+		if err != nil {
+			return nil, err
+		}
+
+		updated, err := r.pauseMachineConfigPool(mcpUpdate, false)
+		if err != nil {
+			return nil, err
+		}
+
+		if updated {
+			return &reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+		return nil, nil
+	}
+
+	// create machine config pool and pause it
+	if mcpMutated != nil && created {
```

Comment:
if we know the mcp needs to be created here, and we're just going to immediately pause it after creation, then why not post the manifest with paused=true to begin with?

so

```
if mcpMutated != nil && needsToBeCreated {
// post a new paused mcp. 
} else {
// get and pause existing mcp
}

```

Reply:
done